fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/mixin/networking/ServerConfigurationNetworkHandlerMixin.java

```diff
@@ -51,14 +81,63 @@ private void initAddon(CallbackInfo ci) {
 		this.addon.lateInit();
 	}
 
-	@Inject(method = "sendConfigurations", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/MinecraftServer;getCombinedDynamicRegistries()Lnet/minecraft/registry/CombinedDynamicRegistries;"))
+	@Inject(method = "sendConfigurations", at = @At("HEAD"), cancellable = true)
 	private void onClientReady(CallbackInfo ci) {
-		this.addon.onClientReady();
+		// Send the initial channel registration packet
+		if (this.addon.startConfiguration()) {
+			assert currentTask == null;
+			ci.cancel();
+			return;
+		}
+
+		// Ready to start sending packets
+		if (!sentConfiguration) {
+			this.addon.preConfiguration();
+			sentConfiguration = true;
+			earlyTaskExecution = true;
+		}
+
+		// Run the early tasks
+		if (earlyTaskExecution) {
+			if (pollEarlyTasks()) {
+				ci.cancel();
+				return;
+			} else {
+				earlyTaskExecution = false;
+			}
+		}
+
+		// All early tasks should have been completed
+		assert currentTask == null;
+		assert tasks.isEmpty();
+
+		// Run the vanilla tasks.
+		this.addon.configuration();
```

Comment:
The events seems to be missing on the in-play reconfiguration code path, which never calls `sendConfigurations` (used by login to configuration only)

Reply:
Ill check this, im fairly sure it does work when reconfiguring.

Reply:
Oh yeah, you are correct. I tottaly missed that `queueJoinWorldTask` method 👍 As its only called from the debug command right now I think its ok to leave/handle at a later date. Its a little unclear to me how this is intended to be used.